src/testers.jl

```diff
@@ -205,6 +205,10 @@ function test_rrule(
         for (accumulated_x̄, x̄_ad, x̄_fd) in zip(accumulated_x̄, x̄s_ad, x̄s_fd)
             if accumulated_x̄ isa Union{Nothing, DoesNotExist}  # then we marked this argument as not differentiable # TODO remove once #113
                 @assert x̄_fd === nothing  # this is how `_make_j′vp_call` works
+                x̄_ad isa Zero && error(
+                    "The pullback in the rrule for $f function should use DoesNotExist()" *
+                    " rather than Zero() for non-differentiable arguments."
```

Comment:
I kind of want to say non-perturbable or discretely valued inputs.
Non-differentiable feels to vague, and also lime it is a property primilily of the function rather than the input